cmd/s3-gw/app_metrics.go

```diff
@@ -63,18 +72,44 @@ func newStateMetrics() *stateMetrics {
 	}
 }
 
-func (m stateMetrics) register() {
-	prometheus.MustRegister(m.healthCheck)
-}
-
-func (m stateMetrics) unregister() {
+func (m *stateMetrics) unregister() {
 	prometheus.Unregister(m.healthCheck)
 }
 
-func (m stateMetrics) SetHealth(status healthStatus) {
+func (m *stateMetrics) SetHealth(status healthStatus) {
 	m.healthCheck.Set(float64(status))
 }
 
+func (m *stateMetrics) updateHealthStatus() {
+	ctx, cancel := context.WithTimeout(context.Background(), checkNeoFSConnectionTimeOut)
+	defer cancel()
+
+	cl, err := m.p.RawClient()
+	// Only "no healthy client" error is possible.
+	if err != nil {
+		m.SetHealth(healthStatusUnhealthy)
+		return
+	}
+
+	// Check actual connection is really alive.
+	_, err = cl.NetworkInfo(ctx, client.PrmNetworkInfo{})
+	if err != nil && strings.Contains(err.Error(), "connection error") {
```

Comment:
i'd avoid error stringing approach AMAP, it can break at any t. If u need to catch this error, it should be exposed from the package